src/config.js

```diff
@@ -23,18 +25,24 @@ export function applyConfigToArgv({
   for (const option in configObject) {
     // we assume the value was set on the CLI if the default value is
     // not the same as that on the argv object as there is a very rare chance
-    // this happening
+    // of this happening
+    if (camelCase(option) !== option) {
+      throw new UsageError(`Please use camel case to specify ${option} ` +
+        'in the config');
+    }
     const wasValueSetOnCLI = typeof(argv[option]) !== 'undefined' &&
       (argv[option] !== defaultValues[option]);
     if (wasValueSetOnCLI) {
       log.debug(`Favoring CLI: ${option}=${argv[option]} over ` +
         `configuration: ${option}=${configObject[option]}`);
       continue;
     }
-    if (!argv.hasOwnProperty(option)) {
+    if (!argv.hasOwnProperty(option) &&
+      !argv.hasOwnProperty(decamelize(option, '-'))) {
       log.debug(`Ignoring configuration: ${option}=${configObject[option]} ` +
         'because this is an unknown option');
-      continue;
+      throw new UsageError(`The option ${option} is invalid.` +
+        ' Please fix your config and try again.');
```

Comment:
Rather than say "please fix [thing]," I think it's more useful to state exactly what the error was as a matter of fact. I would suggest something more like:
````js
throw new UsageError(
  `The config file at ${configFilename} specified an unknown option: "${option}"`
);
````